### PR TITLE
Keypad

### PR DESCRIPTION
**Pull request by sbenejam**
Added +% and -% buttons functionality. An xx% ammount can be added or subtracted to a channel or a range of channels actual values.
Added Reset All Channels and Reset Channel buttons.

### CODE REVIEW

File: webaccess/res/keypad.html [hunk 1]
```diff
@@ -67,7 +67,9 @@
 var vall=[];
 var chanlvall=[];
 var channels, values, c, v, first, last, step, range;
-
+var reqchan;
+var addval;
+var valors=[];
```
Comment (mcallegari) [inline webaccess/res/keypad.html]: Please remove 'values' above and rename this array into 'values'.

File: webaccess/res/keypad.html [hunk 2]
```diff
@@ -81,6 +83,21 @@
...
+    var msg=ev.data;
+    var msgParams = ev.data.split('|');
+    if (msgParams[0] == "QLC+API") {
+      if (msgParams[1] == "getChannelsValues") {
+        valors=[];
+        for (i = 2; i < msgParams.length; i+=3){
+            console.log(msgParams[i+1]);
+            valors.push(msgParams[i+1]);
+        }
+        console.log(valors);
```
Comment (mcallegari) [inline webaccess/res/keypad.html]: Please remove debug stuff

File: webaccess/res/keypad.html [hunk 3]
```diff
@@ -160,18 +211,18 @@
 // Calls to chval fucntion to evaluate channels numbers and values.
 // Arrange it on an array used later to assign channels values
 // calling setChannel function in a loop
-
-  chanlvall = []
-  chanl = []
-  vall = []
+    console.log(c);
```
Comment (mcallegari) [inline webaccess/res/keypad.html]: Same as above, please remove

File: webaccess/res/keypad.html [hunk 4]
```diff
@@ -187,18 +238,36 @@
       for (var i = vals[0]; i <= vals[1]; i += Math.round(vals[1]/llarg ))
         vall.push(i);
     }
+    for (var i = chans[0]; i <= chans[1]; i += chans[2]){
```
Comment (mcallegari) [inline webaccess/res/keypad.html]: Same as above, please remove

File: webaccess/res/keypad.html [hunk 5]
```diff
@@ -187,18 +238,36 @@
...
     chanlvall = zip(chanl, vall);
     document.getElementById('commInput').value = '';
     // Iterate on array chanvall and set dmx values for every channel number
+    //oldvals=zip(valors);
     for (i = 0, len = chanlvall.length; i  < len; i++)
     {
-      // console.log(chanlvall[i][0],chanlvall[i][1]);
       c = chanlvall[i][0];
       v = chanlvall[i][1];
+      console.log(c, v);
```
Comment (mcallegari) [inline webaccess/res/keypad.html]: Same as above, please remove

**Comment by mcallegari** [comment]
We should also add "title" to the buttons, to have self-explaining tooltips of what each button does.
Since we don't have translations here, english would be just fine

**Comment by sbenejam** [comment]
I also added the title to the non-numeric buttons. ;-)

**Comment by mcallegari** [comment]
Excellent, thanks ! :+1:

**Comment by sbenejam** [comment]
I apologize for the errors in the code indentation.
;-)


2017-03-09 19:30 GMT+01:00 Massimo Callegari <notifications@github.com>:

**Pull request merged**